packages/block-editor/src/components/inserter/quick-inserter.js

```diff
@@ -86,6 +90,14 @@ export default function QuickInserter( {
 	// When clicking Browse All select the appropriate block so as
 	// the insertion point can work as expected.
 	const onBrowseAll = () => {
+		if (
+			isMobileDevice &&
+			closeQuickInserter &&
+			typeof closeQuickInserter === 'function'
+		) {
+			closeQuickInserter();
+		}
```

Comment:
Suggestion: You could use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) here and remove `closeQuickInserter` prop checks.

```suggestion
		if ( isMobileDevice  ) {
			closeQuickInserter?.();
		}
```